### PR TITLE
alexj105: feat offer status in conversation [PURCHASE-2704] [PURCHASE-2795]

### PR DESCRIPTION
**Pull request by alexj105**
I combined these tickets for a simpler review and for implementing some tests for it(fetching data and view part)
The status of a sent offer/s collated between messages.
investigating and implementing some logic from eigen to force

https://artsyproduct.atlassian.net/browse/PURCHASE-2704
https://artsyproduct.atlassian.net/browse/PURCHASE-2795

Screen Shot 2021-07-01 at 3 18 08 PM

Screen Shot 2021-07-02 at 9 49 56 AM

### CODE REVIEW

**Review by xtina-starr — APPROVED** [review]
@alexj105 this looks good to me. I'm pinging @erikdstock & @lilyfromseattle as they are bit more familiar with this code than me.

Also there are a few merge conflicts.

File: src/v2/Apps/Conversation/Components/ConversationMessages.tsx [hunk 1]
```diff
@@ -1,60 +1,133 @@
...
+    )
+    const groupAllMessages = groupMessages(sortedMessages)
+    setMessagesAndEvents(groupAllMessages.reverse())
+  }, [messages, events])
+
+  const relevantEvents = [
+    "CommerceOfferSubmittedEvent",
+    "CommerceOrderStateChangedEvent",
+  ]
+  type DisplayableMessage = OrderUpdate_event | Message_message
```
Comment (erikdstock) [inline src/v2/Apps/Conversation/Components/ConversationMessages.tsx:68]: minor: the `type` here mixed in with code is a little awkward to me. Some may disagree with me though if it is very specific to this section. Non-blocking!

File: src/v2/Apps/Conversation/Components/TimeSince.tsx [hunk 2]
```diff
@@ -85,12 +85,15 @@ export const TimeSince: React.FC<TimeSinceProps> = ({
...
   ...props
 }) => {
+  if (!time) {
+    return
+  }
   return (
     time && (
       <Box {...props}>
-        <Sans size={size} color="black30">
+        <Text variant="small" color="black30">
```
Comment (erikdstock) [inline src/v2/Apps/Conversation/Components/TimeSince.tsx:94]: does this mean the `size` prop is no longer needed?

**Review by erikdstock — COMMENTED** [review]
🥵 lookin pretty good, two minor comments here!

File: src/v2/Apps/Conversation/Components/ConversationMessages.tsx (hunk 1 above)
Comment (alexj105) [reply src/v2/Apps/Conversation/Components/ConversationMessages.tsx]: Thank you for this note! Make a sense!

File: src/v2/Apps/Conversation/Components/TimeSince.tsx (hunk 2 above)
Comment (alexj105) [reply src/v2/Apps/Conversation/Components/TimeSince.tsx:94]: @can I merge after these fixes?

File: src/desktop/lib/deprecated_global_client_setup.tsx [hunk 3]
```diff
@@ -2,7 +2,7 @@
  * NOTE: This file has been deprecated as all new JS code lives in `src/v2`.
  * The following only applies to older apps in src/desktop.
  */
-import "./webpackPublicPath"
```
Comment (jo-rs) [inline src/desktop/lib/deprecated_global_client_setup.tsx:5]: Is this an intended change? 🤔

**Comment by jo-rs** [comment]
Should we be seeing all these non-related commits included here? It makes the diff hard to assess.

**Review by lilyfromseattle — COMMENTED** [review]
I’m having some difficulty reviewing all of the changes, as some appear to be unrelated to the two stories tagged. Do jira tickets exist for the changes unrelated to offer status and outside of the context of conversations?

The work related to the order status updates looks good! I approve that, just unsure about the other changes without context.

In the future, I'd prefer multiple smaller PRs to one big one, 73 changed files is pretty overwhelming to review at once, especially without all the necessary context and linked jira tickets.

**Comment by alexj105** [comment]
> I’m having some difficulty reviewing all of the changes, as some appear to be unrelated to the two stories tagged. Do jira tickets exist for the changes unrelated to offer status and outside of the context of conversations?
> 
> The work related to the order status updates looks good! I approve that, just unsure about the other changes without context.
> 
> In the future, I'd prefer multiple smaller PRs to one big one, 73 changed files is pretty overwhelming to review at once, especially without all the necessary context and linked jira tickets.

@lilyfromseattle @jo-rs omg! Sorry, I used merge(( sorry for it, I was also a bit confused when I see it. It won't happen again :)

**Pull request merged**